### PR TITLE
feat: add price filter

### LINKED ISSUE #343: Price list display in shop and product pages by descending date order
### Problem
With addition of older prices and GDPR prices, shop and product pages information is less usable.
- It would be great to display prices by descending date and not descending input.
- Additionally add other filters (for people who still want to see the latest inputs for example)

### PR DESCRIPTION
**Pull request by dq18**
### What
- With our database of prices growing and huge GDPR import, we need to have different filters to display prices
- Added price ordering by "last created" descending and "date"
- Added price filtering for the last 30 days (can be further improved/reworked)
- Pages concerned: ProductDetail, LocationDetail, UserDetail

### Screenshot
image
### Fixes bug

- Added a small fix to ProductList to avoid doing an API call when going from ProductList page to any other

Closes #343

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -10,11 +10,18 @@ export default {
   PRODUCT_FILTER_LIST: [
     { key: 'hide_price_count_gte_1', value: 'Hide products with prices' },
   ],
+  PRICE_FILTER_LIST: [
+    { key: 'show_last_month', value: 'Show Prices from Last 30 Days' },
```
Comment (raphodn) [inline src/constants.js:14]:
```suggestion
    { key: 'only_last_30d', value: 'Only prices for the last 30 days' },
```

Comment (raphodn) [reply src/constants.js:14]: btw these order / filter labels are not translated (seperate PR)

File: src/views/LocationDetail.vue [hunk 2]
```diff
@@ -62,13 +89,43 @@ export default {
...
+  computed: {
+    locationId() {
+      return this.$route.params.id
+    },
+    getCurrentPriceOrderIcon() {
+      let currentPriceOrder = this.priceOrderList.find(o => o.key === this.priceOrder)
+      return currentPriceOrder ? currentPriceOrder.icon : ''
+    },
+    getPricesParams() {
+      let defaultParams = { location_id: this.locationId, order_by: `${this.priceOrder}`, page: this.locationPricePage }
```
Comment (raphodn) [inline src/views/LocationDetail.vue:114]:
```suggestion
      let defaultParams = { location_id: this.locationId, order_by: this.priceOrder, page: this.locationPricePage }
```
string interpolation not need, right ?

File: src/views/LocationDetail.vue [hunk 3]
```diff
@@ -62,13 +89,43 @@ export default {
...
+      if (this.priceFilter === 'show_last_month') {
+        let oneMonthAgo = new Date()
+        oneMonthAgo.setMonth(oneMonthAgo.getMonth() - 1)
+        defaultParams['date__gte'] = oneMonthAgo.toISOString().split('T')[0]
```
Comment (raphodn) [inline src/views/LocationDetail.vue:118]: in the code elsewhere we used `toISOString().substring(0, 10)`. better ?

**Review by raphodn — APPROVED** [review]
cool ! left only wording/cosmetic comments

File: src/views/LocationDetail.vue [hunk 4]
```diff
@@ -31,6 +31,32 @@
     <v-progress-circular v-if="loading" indeterminate :size="30"></v-progress-circular>
   </h2>
 
+  <v-row v-if="!loading">
+    <v-col>
+      <v-menu>
+        <template v-slot:activator="{ props }">
+          <v-btn v-bind="props" size="small" class="mr-2" prepend-icon="mdi-filter-variant" :active="!!priceFilter">{{ $t('BrandDetail.Filter') }}</v-btn>
```
[the remark this answers is not in the document]
Comment (dq18) [reply src/views/LocationDetail.vue:38]: I forgot to change this! We should have a common translation indeed

File: src/constants.js (hunk 1 above)
Comment (dq18) [reply src/constants.js:14]: Yes, I wanted to discuss on this first since it would mean adding translation in constant.js

Comment (raphodn) [reply src/constants.js:14]: Or doing something similar as `routes.js` ?

File: src/views/LocationDetail.vue (hunk 3 above)
Comment (dq18) [reply src/views/LocationDetail.vue:118]: Tested on a 1000000 loop
split: 318.093994140625 ms
substring: 3.641845703125 ms
split: 264.632080078125 ms
substring: 0.674072265625 ms

substring is indeed faster (300 nano second compared to 1ns). But both are still super fast. I guess it is down to readability. Both are fine by me

File: src/views/LocationDetail.vue (hunk 2 above)
Comment (dq18) [reply src/views/LocationDetail.vue:114]: Indeed, correcting it

File: src/views/LocationDetail.vue (hunk 3 above)
Comment (raphodn) [reply src/views/LocationDetail.vue:118]: Wasn't thinking about speed. Just uniformity 👍

File: src/views/LocationDetail.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (dq18) [reply src/views/LocationDetail.vue:38]: It seems there is a "Translation Memory" feature:
https://support.crowdin.com/enterprise/common-questions-translators/#can-i-use-project-translation-memory-where-do-i-find-it
So if I rename the key, Crowdin should add all translations from memory to the renamed key.

File: src/constants.js [hunk 5]
```diff
@@ -10,11 +10,18 @@ export default {
...
+  ],
   ORDER_BY_PARAM: 'order_by',
   PRODUCT_ORDER_BY_LIST: [
     { key: '-unique_scans_n', value: 'Number of scans', icon: 'mdi-barcode-scan' },
     { key: '-price_count', value: 'Number of prices', icon: 'mdi-tag-multiple-outline' },
   ],
+  PRICE_ORDER_BY_LIST: [
+    { key: '-created', value: 'Last created', icon: 'mdi-calendar-plus' },
```
[the remark this answers is not in the document]
Comment (dq18) [reply src/constants.js]: Done

File: src/constants.js [hunk 6]
```diff
@@ -10,11 +10,18 @@ export default {
...
+    { key: '-date', value: 'Date (Newest First)', icon: 'mdi-calendar' },
```
[the remark this answers is not in the document]
Comment (dq18) [reply src/constants.js]: Done

File: src/views/LocationDetail.vue [hunk 7]
```diff
@@ -31,6 +31,32 @@
...
+        <v-list>
+          <v-list-item :slim="true" v-for="filter in priceFilterList" :key="filter.key" :prepend-icon="(priceFilter === filter.key) ? 'mdi-check-circle' : 'mdi-circle-outline'" :active="priceFilter === filter.key" @click="togglePriceFilter(filter.key)">
+            {{ filter.value }}
+          </v-list-item>
+        </v-list>
+      </v-menu>
+
+      <v-menu>
+        <template v-slot:activator="{ props }">
+          <v-btn v-bind="props" size="small" prepend-icon="mdi-arrow-down" :append-icon="getCurrentPriceOrderIcon"  :active="!!priceOrder">{{ $t('BrandDetail.Order') }}</v-btn>
```
[the remark this answers is not in the document]
Comment (dq18) [reply src/views/LocationDetail.vue]: Fixed

File: src/constants.js (hunk 1 above)
Comment (dq18) [reply src/constants.js]: Will create another PR

**Pull request merged**